packages/mason_cli/lib/src/commands/bundle.dart

```diff
@@ -46,15 +64,45 @@ class BundleCommand extends MasonCommand {
 
   @override
   Future<int> run() async {
-    if (results.rest.isEmpty) {
-      usageException('path to the brick template must be provided');
+    final source = results['source'] as String;
+
+    late final rest = results.rest.first;
+
+    final Brick brick;
+    if (source == 'git') {
+      if (results.rest.isEmpty) {
+        usageException('A repository url must be provided');
+      }
+      brick = Brick(
+        location: BrickLocation(
+          git: GitPath(
+            rest,
+            path: results['git-path'] as String?,
+            ref: results['git-ref'] as String?,
+          ),
+        ),
+      );
+    } else if (source == 'hosted') {
+      if (results.rest.isEmpty) {
+        usageException('A brick name must be provided');
+      }
+      brick = Brick(name: rest, location: const BrickLocation(version: 'any'));
```

Comment:
should users be able to specify a specific version to bundle?

Reply:
As the `add` command doesn't allow specific versions, I didn't want to overstep the implementation here.